src/Queue/DoctrineQueue.php

```diff
@@ -123,8 +135,39 @@ public function pop(array $options = []): ?JobInterface
         // First run garbage collection
         $this->purge();
 
-        $conn = $this->connection;
-        $conn->beginTransaction();
+        if (self::DATABASE_PLATFORM_POSTGRES == $this->connection->getDatabasePlatform()->getName()) {
```

Comment:
I think we can test more easily if we make a property on this class that determines the pop strategy. And obviously the factory of this queue will only set the MySQL-style pop if MySQL is used.

Furthermore: I think it makes more sense to only enable the mysql pop style for MySQL, and the postgres style as default. As it could well be that MySQL is the exception.

Reply:
We have another "if postgres" check in push()... how do we want to handle this?

What other Doctrine-supported engines are there? SQLite doesnt support "SELECT ... FOR UPDATE", so we'd have to use the MySQL-pop too. Dont know if MSSQL, ... correctly support this.

From my perspective the mysql variant is the most portable because it uses just simple UPDATE and SELECTs without any locking or transactions. PostgreSQL has an optimization which we cant assume for all other db engines.

What do you think?